### PR TITLE
Add to conditional card le,ge conditions

### PR DESCRIPTION
**Pull request by jedi7**
Lovelace conditional card now supports conditions operators
* operator '=='  as equal
* operator '!='  as not equal
* operator '>'  as greater
* operator '<'  as less
* operator '>='  as greater or equal
* operator '<='  as less or equal

UI editor is not supported now

Alternative (simpler) version added here https://github.com/home-assistant/frontend/pull/7688

## Proposed change

Lovelace conditional card supports only 'equal' or 'not equal', so this patch adds operator where user can specify operation.
UI editor is not supported.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
type: conditional
  - entity: sensor.elektromer_l1
    operator: '<'
    value: 10
  - entity: sensor.elektromer_l1
    state: 20
card:
...
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/15302
- Related discussions: 
https://community.home-assistant.io/t/conditional-card-with-and-as-conditions/182771
https://community.home-assistant.io/t/support-numeric-state-for-conditional-card-in-lovelace/79563

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -2384,6 +2384,8 @@
               "card": "Card",
               "state_equal": "State is equal to",
               "state_not_equal": "State is not equal to",
+              "state_le": "State is lower or equal to",
+              "state_ge": "State is greather or equal to",
```
Comment (spacegaier) [inline src/translations/en.json:2388]: Should be "greater"

Comment (jedi7) [reply src/translations/en.json]: fixed, thanks!

**Comment by bramkragten** [comment]
We should align this with state filter <https://www.home-assistant.io/lovelace/entity-filter/#options-for-state_filter>

**Comment by jedi7** [comment]
uff, that is quite out of my knowledge for now. It will be also breaking change.

**Comment by bramkragten** [comment]
Doesn't have to be breaking, we can either translate the old config to the new, or only apply the operator to `state`

**Comment by jedi7** [comment]
What about to merge the Conditional Card and Entity Filter Card together?
Make it sense?

possible configuration example:
```yaml
type: entity-filter
entities:
  - device_tracker.demo_paulus
  - device_tracker.demo_anne_therese
  - device_tracker.demo_home_boy
state_filter:
  - operator: "!="
    value: home
  - operator: "!="
    value: work
card:
  type: entities
  title: Who's Running Errands   # optional
  entities:   # optional, when not present, the entities from filter will be used
    - device_tracker.demo_paulus
    - cover.kitchen_window
    - group.kitchen
    - lock.kitchen_door
    - light.bed_light
```

**Comment by jedi7** [comment]
So, should I try to implement the merged configuration?

**Comment by bramkragten** [comment]
No, I don't think that makes things clearer.

For now, replace `state_ge`, `state_le` etc by a new `operator` property and keep the `state` for the value.

**Comment by jedi7** [comment]
Can you show me example please? 
Someting like that?

`type: conditional
conditions:
  - entity: sensor.elektromer_l1
    operator: '!='
    value: 10
  - entity: sensor.elektromer_l1
    operator: '<='
    value: 20
card:
...`

**Comment by bramkragten** [comment]
```
type: conditional
conditions:
  - entity: sensor.bed_light
    state: "10"
    operator: "=>"
  - entity: switch.decorative_lights
    state_not: "off"
card:
  type: entities
  entities:
    - device_tracker.demo_paulus
    - cover.kitchen_window
    - group.kitchen
    - lock.kitchen_door
    - light.bed_light
```

**Comment by jedi7** [comment]
Are you sure? It does not looks too much consistent. Btw In entity filter is used combination "operator", "value". And in condition card "state_" + operator postfix. Imho we should select one solution and keep it :)

**Comment by bramkragten** [comment]
Yes, but we are not allowed to make breaking changes, so we should support both.

**Comment by jedi7** [comment]
Ok it was not so hard :)
Updated to this configuration:
I used pair "operator" and "value" for consistency (and it also make more sense)
If it will be ok, I can try update also the GUI editor and documentation.

```yaml
type: conditional
conditions:
  - entity: sensor.elektromer_l1
    operator: '<'
    value: 10
  - entity: sensor.elektromer_l1
    state: 20
card:
...
```

File: src/panels/lovelace/common/validate-condition.ts [hunk 2]
```diff
@@ -15,13 +17,50 @@ export function checkConditionsMet(
...
+      return true;
+    }
 
-    return c.state ? state === c.state : state !== c.state_not;
+    if (!c.value) {
+      // error state must be defined when operator is defined too
+      return false;
+    }
+
+    if (c.operator === "==") {
```
Comment (bramkragten) [inline src/panels/lovelace/common/validate-condition.ts:35]: Can we reuse evaluateFilter <https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/common/evaluate-filter.ts#L3>?

Comment (jedi7) [reply src/panels/lovelace/common/validate-condition.ts:35]: I can try it. But how are there handled numbers? I do not see conversion from string to number. Or it is not needed?

Comment (bramkragten) [reply src/panels/lovelace/common/validate-condition.ts:35]: It should already be a number? We should not need to convert it

Comment (jedi7) [reply src/panels/lovelace/common/validate-condition.ts]: done, thanks :)

**Comment by bramkragten** [comment]
Looks good, we need to update the documentation for this

**Comment by jedi7** [comment]
documentation updated (but not sure about the section deprecated config)

**Comment by jedi7** [comment]
just for note, here is the PR for documentation: https://github.com/home-assistant/home-assistant.io/pull/15302

**Comment by frenck** [comment]
Just picked up on this because of the documentation PR. I think changing `state` to `value` is really unnecessary, confusing compared to how it is named compared to the backend.

IMHO, there is no need to change that or divert from the naming we are already used to.

**Comment by jedi7** [comment]
@frenck Can looks confusing, but there was reason for it. Please see at https://www.home-assistant.io/lovelace/entity-filter/ where are used operators too. The function for validation is reused from there. So the change was to have consistent naming across these filter/condition cards.

Also I'm not sure how implement for example this configuration:
```yaml
conditions:
  - entity: sensor.elektromer_l1
    operator: '<'
    state_not: 10
```
So that is reason to use `value` than `state`.

Not sure how it is done in backed. Maybe some refactoring will be needed, but I think it will take time, and will be cool to have at least something for now.

**Comment by frenck** [comment]
> and will be cool to have at least something for now.

☝️ That never ever can be a reason to do it. In a project of this size, adding something means maintaining it. Changing is later is a pain.

**Comment by jedi7** [comment]
So how the proposed configuration should look?

**Comment by jedi7** [comment]
Anyway if I understand, now we have three versions of conditions:
1) operator, value (in filter entity card)
2) state, state_not (in condition card)
3) above, below, ... (in backend)

I personally prefer the mathematics schema (operator,value). So I wanted to reduce the versions to 2 (frontend operator, value and backed will still have above,below,..)

What you think?
Because to unify all, it means to do (breaking) change the filter entity card and condition card.

**Comment by jedi7** [comment]
Guys, so what I should do? I can change it back to `state`, `state_not`, and `below`, 'above' (I had previously state_le, state_ge). But then I cannot reuse the validation from entity filter.

**Comment by frenck** [comment]
I have no opinion on the end result at this point. If you want my opinion on the right path for the broader scope of things; I'll have to sit down and look at the options and think them through.

The only reason I pulled in here, was because the configurations differ, which I consider a bad thing.

**Comment by jedi7** [comment]
@bramkragten What you think? Should it be `state` , `above`, .. etc, or the operator, or both? I want to finish this PR :)

**Comment by bramkragten** [comment]
I kinda agree with @frenck, that it is confusing with the backend, but that would mean changing the config schema for the filter entity card too. And it would limit the options...

**Comment by frenck** [comment]
Doesn't have to be... we don't have to match the backend, the backend can match up as well. (No right answer but worth the look?)

**Comment by jedi7** [comment]
ok guys, what about to have both? Support operators and the named operations? If I understand, for some situations we need simple config and for another a little complex.
The named operations will be changeable by UI editor.

**Comment by bramkragten** [comment]
More options, means harder to maintain, so I'm not a fan of that.

**Comment by balloob** [comment]
Complex config doesn't belong in Home Assistant. That's for custom cards. So maybe we should just close this?

**Comment by jedi7** [comment]
@balloob  not sure what you mean by "just close this".

So my propose is: I will remove the `operator` and just add the `above`, `below`. So it will match with the current logic in Condition card and logic in backend.
Is it ok? I do not want to remake it 10x ;-)

**Comment by jedi7** [comment]
@frenck @bramkragten Guys, do you will agree with adding just keywords `above`, `below`? So we can move on. Thanks

**Comment by jedi7** [comment]
hmm, it seems nobody want it :(

**Comment by marecabo** [comment]
I would be very glad to see an `above`/`below` condition added to the conditional card.

**Comment by larswnd** [comment]
+1

**Comment by JDunnio** [comment]
any progress on this? Would love to see this for displaying notifications on my home screen.
Eq or ne doesnt cut it all the times and takes a extra template sensor.

**Comment by spacegaier** [comment]
> 
> 
> any progress on this? Would love to see this for displaying notifications on my home screen.
> Eq or ne doesnt cut it all the times and takes a extra template sensor.

As you can see, this PR has been closed, but there is a similar alternative one that is being considered: https://github.com/home-assistant/frontend/pull/7688

**Pull request closed without merging**